### PR TITLE
Omniauth test

### LINKED ISSUE #21: Omniauth Testing

### PR DESCRIPTION
**Pull request by Stearnzy**
Description:  
* Omniauth testing implemented.
* Adds missing function for saving user email to database upon login.

Fixes #21 

* [x] All tests passing
* [x] Lint code
* [x] Checked local host
* [ ] Updated README

### CODE REVIEW

File: spec/factories/user_omniauth.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+FactoryBot.define do
+  factory :user do
+    name { Faker::Name.name }
+    email { Faker::Internet.email }
+    uid { Faker::Number.within(range: 100_000..999_999) }
+    token { Faker::Number.within(range: 100_000..999_999) }
+  end
+
+  factory :omniauth_mock_user, parent: :user do
```
Comment (dcoleman21) [inline spec/factories/user_omniauth.rb:11]: Nice set up here.

**Review by dcoleman21 — APPROVED** [review]
From what I know about OmniAuth/OAuth, this looks good!

File: spec/features/welcome/welcome_spec.rb [hunk 2]
```diff
@@ -6,18 +6,49 @@
...
       expect(page).to have_link('Login with Google')
       expect(page).to have_content(app_description)
       expect(page).to have_link('Privacy Policy')
     end
 
-    xit "starts the oauth process when you click on the 'Login with Google' link" do
-      visit '/'
-      click_link('Login with Google')
-      expect(current_path).to eq('/auth/google_oath2')
+    it 'Google OAuth logs in new user' do
```
Comment (Oxalisviolacea) [inline spec/features/welcome/welcome_spec.rb:17]: Nice test!

**Review by Oxalisviolacea — APPROVED** [review]

File: spec/factories/user_omniauth.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'faker'
+
+FactoryBot.define do
```
Comment (Callbritton) [inline spec/factories/user_omniauth.rb:3]: Love to see we implemented Factorybot and Faker here!

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -71,4 +71,44 @@
       with.library :rails
     end
   end
+
+  def stub_omniauth
+    OmniAuth.config.test_mode = true
```
Comment (Callbritton) [inline spec/rails_helper.rb:76]: Great job with this!

**Pull request merged**